### DISCUSSION TITLE
Ability to easily toggle sketchybar

### DISCUSSION THREAD

**Discussion by rajeshmajumdar** [FelixKratz/SketchyBar#349] [Ideas]
Recently I installed dmenu-mac, and it's simply does what it suppose to do. Except the bar stays on top of the menu. Same goes when I use apps like illustrator, for that I need to use the default bar many options are there. But it stays on top of all that, it would be great if in the config file itself I could add some kind of blacklist to auto show/hide the bar.
Since sketchybar already allows drawings to be off through command line, and it also has the capability of knowing which apps we are currently running. In theory this should be possible to implement.

**Comment by FelixKratz** [comment]
You can script something like this like so:

sketchybarrc
```bash
sketchybar --add item toggler left \
           --set toggler drawing=off \
                         updates=on \
                         script="$PLUGIN_DIR/toggle.sh" \
           --subscribe toggler front_app_switched
```

toggle.sh
```bash
#!/bin/sh

case "$INFO" in
  "Finder") sketchybar --bar hidden=on
    ;;
  *) sketchybar --bar hidden=off
esac
```

**Reply by marcelmanz** [reply]
thanks!

**Comment by matto80** [comment]
Hi Felix,
does this work with the lua setup also? Since it relies on many more files than just sketchybarrc
Thanks,
maddie

**Reply by hsulab** [reply]
You can add something like this:

```
front_app:subscribe("front_app_switched", function(env)
  front_app:set({ label = { string = env.INFO } })
  if env.INFO == "Finder" then
    sbar.bar({ hidden = "off" })
  else
    sbar.bar({ hidden = "on" })
  end
end)
```

**Comment by markhagemann** [comment]
Does anyone have a suggestion for toggling this to appear on a space change or focus change and then hiding after 5 seconds or so?
